xdsl/backend/riscv/register_allocation.py

```diff
@@ -22,10 +22,12 @@
 from xdsl.utils.exceptions import DiagnosticException
 
 
-def gather_allocated(func: riscv_func.FuncOp) -> set[RISCVRegisterType]:
+def gather_allocated(
+    func: riscv_func.FuncOp,
+) -> set[IntRegisterType | FloatRegisterType]:
     """Utility method to gather already allocated registers"""
 
-    allocated: set[RISCVRegisterType] = set()
+    allocated: set[IntRegisterType | FloatRegisterType] = set()
 
```

Comment:
Why this change?

Reply:
I'm not 100% sure it was necessary, but the type variable is invariant, which makes things a little weird, now all the types match exactly.